cmd/clusterctl/cmd/upgrade.go

```diff
@@ -16,4 +16,135 @@ limitations under the License.
 
 package cmd
 
-// TODO: upgrade command
+import (
+	"fmt"
+	"os"
+	"sort"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/pkg/client"
+)
+
+var upgradeCmd = &cobra.Command{
+	Use:   "upgrade",
+	Short: "Upgrades Cluster API providers in a management cluster",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return cmd.Help()
+	},
+}
+
+type upgradePlanOptions struct {
+	kubeconfig string
+}
+
+var up = &upgradePlanOptions{}
+
+var upgradePlanCmd = &cobra.Command{
+	Use:   "plan",
+	Short: "Provide a list of recommended target versions for upgrading Cluster API providers in a management cluster",
+	Long: LongDesc(`
+		The upgrade plan command provides a list of recommended target versions for upgrading Cluster API providers in a management cluster.
+		
+		The providers are grouped into management groups, each one defining a set of providers that should be supporting the same cluster API version
+		in order to guarantee the proper functioning of the management cluster.
+
+		Then, for each provider in a management group, the following upgrade options are provided:
+		- The latest patch release for the current Cluster API version.
+		- The latest patch release for the next Cluster API version, if available.`),
+
+	Example: Examples(`
+		# Gets the recommended target versions for upgrading Cluster API providers.
+		clusterctl upgrade plan`),
+
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return runUpgradePlan()
+	},
+}
+
+func init() {
+	upgradePlanCmd.Flags().StringVarP(&up.kubeconfig, "kubeconfig", "", "", "Path to the kubeconfig file to use for accessing the management cluster. If empty, default rules for kubeconfig discovery will be used")
+
+	upgradeCmd.AddCommand(upgradePlanCmd)
+
+	RootCmd.AddCommand(upgradeCmd)
+}
+
+func runUpgradePlan() error {
+	c, err := client.New(cfgFile)
+	if err != nil {
+		return err
+	}
+
+	//TODO: switch to klog as soon as https://github.com/kubernetes-sigs/cluster-api/pull/2150 merge
```

Comment:
Seems we can switch now?

Reply:
Unfortunately no, because we switched to the global logger approach and this is not yet merged. Waiting for https://github.com/kubernetes-sigs/cluster-api/pull/2191

Reply:
Since 2191 is blocked on this PR, we can create an issue to track this work or I'm assuming @fabriziopandini will be submitting a PR to refactor the upgrade cmd to use the global logger.

Reply:
He's going to update 2191 after this goes in